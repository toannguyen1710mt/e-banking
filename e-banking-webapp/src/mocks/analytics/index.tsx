export const ANALYTICS_DATA = [
  {
    title: 'Total Income',
    subtitle: 'Last 30 days income',
    isPositive: true,
    amount: 670000,
    percentageChange: 8,
  },
  {
    title: 'Total Expense',
    subtitle: 'Last 30 days income',
    isPositive: true,
    amount: 270000,
    percentageChange: 8,
  },
  {
    title: 'Total Investment',
    subtitle: 'Last 30 days income',
    isPositive: true,
    amount: 170000,
    percentageChange: 8,
  },
];
