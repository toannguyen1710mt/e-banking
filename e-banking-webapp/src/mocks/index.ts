export * from './options';
export * from './card';
export * from './charts';
export * from './transaction';
export * from './table';
export * from './transferConfirmData';
export * from './target';
export * from './analytics';
export * from './auth';
export * from './ServiceCard';
export * from './formMocks';
