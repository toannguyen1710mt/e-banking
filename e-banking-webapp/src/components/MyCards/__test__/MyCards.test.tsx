import { fireEvent, render } from '@testing-library/react';

// Mock data
import { MOCK_DATA_USER, MOCK_DATA_USER_EMPTY_PROPS } from '@/mocks';

// Components
import { MyCards } from '..';

describe('MyCards component', () => {
  it('Should render snapshot correctly.', () => {
    expect(
      render(
        <MyCards accounts={MOCK_DATA_USER.accounts} onCardSelect={jest.fn()} />,
      ),
    ).toMatchSnapshot();
  });

  it('Should render snapshot with accounts is empty.', () => {
    expect(render(<MyCards onCardSelect={jest.fn()} />)).toMatchSnapshot();
  });

  it('Should renders cards correctly with empty type and accountNumber', () => {
    const { asFragment } = render(
      <MyCards
        accounts={MOCK_DATA_USER_EMPTY_PROPS.accounts}
        onCardSelect={jest.fn()}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should handle select card.', () => {
    const mockOnCardSelect = jest.fn();

    const { getByLabelText } = render(
      <MyCards
        accounts={MOCK_DATA_USER.accounts}
        onCardSelect={mockOnCardSelect}
      />,
    );

    fireEvent.click(getByLabelText('card-item-1'));

    expect(mockOnCardSelect).toHaveBeenCalledWith(MOCK_DATA_USER.accounts?.[1]);
  });
});
