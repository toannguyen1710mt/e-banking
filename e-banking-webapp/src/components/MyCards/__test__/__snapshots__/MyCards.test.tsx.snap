// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCards component Should render snapshot correctly. 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-wrap gap-4"
      >
        <div
          aria-label="card-item-0"
          class="hover:cursor-pointer"
        >
          <div
            class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex h-[88px] w-full min-w-[111px] flex-col justify-between rounded-[6px] border-0 p-2 bg-secondary-300"
            tabindex="-1"
          >
            <div
              class="flex gap-3 p-0"
            >
              <div
                class="flex flex-1 items-center justify-between"
              >
                <p
                  class="text-foreground-200 font-medium text-xs"
                >
                  Main
                </p>
                <svg
                  class="w-[25px] h-[15px]"
                  fill="none"
                  height="35"
                  viewBox="0 0 59 35"
                  width="59"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M41.0762 34.2205C50.791 34.2205 58.6663 26.723 58.6663 17.4744C58.6663 8.22576 50.791 0.728271 41.0762 0.728271C36.7224 0.728271 32.738 2.23419 29.6664 4.72876C26.5948 2.23426 22.6104 0.72838 18.2566 0.72838C8.54186 0.72838 0.666504 8.22586 0.666504 17.4745C0.666504 26.7231 8.54186 34.2206 18.2566 34.2206C22.6105 34.2206 26.5949 32.7147 29.6665 30.2201C32.7381 32.7146 36.7224 34.2205 41.0762 34.2205Z"
                    fill="#ED0006"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M29.6665 30.2197C33.4484 27.1482 35.8465 22.5778 35.8465 17.4742C35.8465 12.3707 33.4484 7.80033 29.6665 4.72879C32.7381 2.23413 36.7226 0.728149 41.0766 0.728149C50.7913 0.728149 58.6667 8.22563 58.6667 17.4742C58.6667 26.7229 50.7913 34.2203 41.0766 34.2203C36.7226 34.2203 32.7381 32.7144 29.6665 30.2197Z"
                    fill="#F9A000"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M29.6668 30.2192C33.4488 27.1476 35.8469 22.5773 35.8469 17.4737C35.8469 12.3702 33.4488 7.79981 29.6668 4.72827C25.8849 7.79981 23.4868 12.3702 23.4868 17.4737C23.4868 22.5773 25.8849 27.1476 29.6668 30.2192Z"
                    fill="#FF5E00"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
            <div
              class="flex items-center justify-between"
            >
              <p
                class="text-foreground-200 font-medium text-2xs opacity-100"
              >
                .... 
                9208
              </p>
              <p
                class="text-foreground-200 font-medium text-2xs opacity-100"
              />
            </div>
          </div>
        </div>
        <div
          aria-label="card-item-1"
          class="hover:cursor-pointer"
        >
          <div
            class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex h-[88px] w-full min-w-[111px] flex-col justify-between rounded-[6px] border-0 p-2 bg-background-500"
            tabindex="-1"
          >
            <div
              class="flex gap-3 p-0"
            >
              <div
                class="flex flex-1 items-center justify-between"
              >
                <p
                  class="text-foreground-100 font-medium text-xs"
                >
                  Checkings
                </p>
                <svg
                  class="w-[25px] h-[15px]"
                  fill="none"
                  height="35"
                  viewBox="0 0 59 35"
                  width="59"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M41.0762 34.2205C50.791 34.2205 58.6663 26.723 58.6663 17.4744C58.6663 8.22576 50.791 0.728271 41.0762 0.728271C36.7224 0.728271 32.738 2.23419 29.6664 4.72876C26.5948 2.23426 22.6104 0.72838 18.2566 0.72838C8.54186 0.72838 0.666504 8.22586 0.666504 17.4745C0.666504 26.7231 8.54186 34.2206 18.2566 34.2206C22.6105 34.2206 26.5949 32.7147 29.6665 30.2201C32.7381 32.7146 36.7224 34.2205 41.0762 34.2205Z"
                    fill="#ED0006"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M29.6665 30.2197C33.4484 27.1482 35.8465 22.5778 35.8465 17.4742C35.8465 12.3707 33.4484 7.80033 29.6665 4.72879C32.7381 2.23413 36.7226 0.728149 41.0766 0.728149C50.7913 0.728149 58.6667 8.22563 58.6667 17.4742C58.6667 26.7229 50.7913 34.2203 41.0766 34.2203C36.7226 34.2203 32.7381 32.7144 29.6665 30.2197Z"
                    fill="#F9A000"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M29.6668 30.2192C33.4488 27.1476 35.8469 22.5773 35.8469 17.4737C35.8469 12.3702 33.4488 7.79981 29.6668 4.72827C25.8849 7.79981 23.4868 12.3702 23.4868 17.4737C23.4868 22.5773 25.8849 27.1476 29.6668 30.2192Z"
                    fill="#FF5E00"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
            <div
              class="flex items-center justify-between"
            >
              <p
                class="text-foreground-100 font-medium text-2xs opacity-50"
              >
                .... 
                9888
              </p>
              <p
                class="text-foreground-100 font-medium text-2xs opacity-50"
              />
            </div>
          </div>
        </div>
        <div
          aria-label="card-item-2"
          class="hover:cursor-pointer"
        >
          <div
            class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex h-[88px] w-full min-w-[111px] flex-col justify-between rounded-[6px] border-0 p-2 bg-background-500"
            tabindex="-1"
          >
            <div
              class="flex gap-3 p-0"
            >
              <div
                class="flex flex-1 items-center justify-between"
              >
                <p
                  class="text-foreground-100 font-medium text-xs"
                >
                  Savings
                </p>
                <svg
                  class="w-[25px] h-[15px]"
                  fill="none"
                  height="35"
                  viewBox="0 0 59 35"
                  width="59"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M41.0762 34.2205C50.791 34.2205 58.6663 26.723 58.6663 17.4744C58.6663 8.22576 50.791 0.728271 41.0762 0.728271C36.7224 0.728271 32.738 2.23419 29.6664 4.72876C26.5948 2.23426 22.6104 0.72838 18.2566 0.72838C8.54186 0.72838 0.666504 8.22586 0.666504 17.4745C0.666504 26.7231 8.54186 34.2206 18.2566 34.2206C22.6105 34.2206 26.5949 32.7147 29.6665 30.2201C32.7381 32.7146 36.7224 34.2205 41.0762 34.2205Z"
                    fill="#ED0006"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M29.6665 30.2197C33.4484 27.1482 35.8465 22.5778 35.8465 17.4742C35.8465 12.3707 33.4484 7.80033 29.6665 4.72879C32.7381 2.23413 36.7226 0.728149 41.0766 0.728149C50.7913 0.728149 58.6667 8.22563 58.6667 17.4742C58.6667 26.7229 50.7913 34.2203 41.0766 34.2203C36.7226 34.2203 32.7381 32.7144 29.6665 30.2197Z"
                    fill="#F9A000"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M29.6668 30.2192C33.4488 27.1476 35.8469 22.5773 35.8469 17.4737C35.8469 12.3702 33.4488 7.79981 29.6668 4.72827C25.8849 7.79981 23.4868 12.3702 23.4868 17.4737C23.4868 22.5773 25.8849 27.1476 29.6668 30.2192Z"
                    fill="#FF5E00"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
            <div
              class="flex items-center justify-between"
            >
              <p
                class="text-foreground-100 font-medium text-2xs opacity-50"
              >
                .... 
                9357
              </p>
              <p
                class="text-foreground-100 font-medium text-2xs opacity-50"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-wrap gap-4"
    >
      <div
        aria-label="card-item-0"
        class="hover:cursor-pointer"
      >
        <div
          class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex h-[88px] w-full min-w-[111px] flex-col justify-between rounded-[6px] border-0 p-2 bg-secondary-300"
          tabindex="-1"
        >
          <div
            class="flex gap-3 p-0"
          >
            <div
              class="flex flex-1 items-center justify-between"
            >
              <p
                class="text-foreground-200 font-medium text-xs"
              >
                Main
              </p>
              <svg
                class="w-[25px] h-[15px]"
                fill="none"
                height="35"
                viewBox="0 0 59 35"
                width="59"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M41.0762 34.2205C50.791 34.2205 58.6663 26.723 58.6663 17.4744C58.6663 8.22576 50.791 0.728271 41.0762 0.728271C36.7224 0.728271 32.738 2.23419 29.6664 4.72876C26.5948 2.23426 22.6104 0.72838 18.2566 0.72838C8.54186 0.72838 0.666504 8.22586 0.666504 17.4745C0.666504 26.7231 8.54186 34.2206 18.2566 34.2206C22.6105 34.2206 26.5949 32.7147 29.6665 30.2201C32.7381 32.7146 36.7224 34.2205 41.0762 34.2205Z"
                  fill="#ED0006"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M29.6665 30.2197C33.4484 27.1482 35.8465 22.5778 35.8465 17.4742C35.8465 12.3707 33.4484 7.80033 29.6665 4.72879C32.7381 2.23413 36.7226 0.728149 41.0766 0.728149C50.7913 0.728149 58.6667 8.22563 58.6667 17.4742C58.6667 26.7229 50.7913 34.2203 41.0766 34.2203C36.7226 34.2203 32.7381 32.7144 29.6665 30.2197Z"
                  fill="#F9A000"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M29.6668 30.2192C33.4488 27.1476 35.8469 22.5773 35.8469 17.4737C35.8469 12.3702 33.4488 7.79981 29.6668 4.72827C25.8849 7.79981 23.4868 12.3702 23.4868 17.4737C23.4868 22.5773 25.8849 27.1476 29.6668 30.2192Z"
                  fill="#FF5E00"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
          <div
            class="flex items-center justify-between"
          >
            <p
              class="text-foreground-200 font-medium text-2xs opacity-100"
            >
              .... 
              9208
            </p>
            <p
              class="text-foreground-200 font-medium text-2xs opacity-100"
            />
          </div>
        </div>
      </div>
      <div
        aria-label="card-item-1"
        class="hover:cursor-pointer"
      >
        <div
          class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex h-[88px] w-full min-w-[111px] flex-col justify-between rounded-[6px] border-0 p-2 bg-background-500"
          tabindex="-1"
        >
          <div
            class="flex gap-3 p-0"
          >
            <div
              class="flex flex-1 items-center justify-between"
            >
              <p
                class="text-foreground-100 font-medium text-xs"
              >
                Checkings
              </p>
              <svg
                class="w-[25px] h-[15px]"
                fill="none"
                height="35"
                viewBox="0 0 59 35"
                width="59"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M41.0762 34.2205C50.791 34.2205 58.6663 26.723 58.6663 17.4744C58.6663 8.22576 50.791 0.728271 41.0762 0.728271C36.7224 0.728271 32.738 2.23419 29.6664 4.72876C26.5948 2.23426 22.6104 0.72838 18.2566 0.72838C8.54186 0.72838 0.666504 8.22586 0.666504 17.4745C0.666504 26.7231 8.54186 34.2206 18.2566 34.2206C22.6105 34.2206 26.5949 32.7147 29.6665 30.2201C32.7381 32.7146 36.7224 34.2205 41.0762 34.2205Z"
                  fill="#ED0006"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M29.6665 30.2197C33.4484 27.1482 35.8465 22.5778 35.8465 17.4742C35.8465 12.3707 33.4484 7.80033 29.6665 4.72879C32.7381 2.23413 36.7226 0.728149 41.0766 0.728149C50.7913 0.728149 58.6667 8.22563 58.6667 17.4742C58.6667 26.7229 50.7913 34.2203 41.0766 34.2203C36.7226 34.2203 32.7381 32.7144 29.6665 30.2197Z"
                  fill="#F9A000"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M29.6668 30.2192C33.4488 27.1476 35.8469 22.5773 35.8469 17.4737C35.8469 12.3702 33.4488 7.79981 29.6668 4.72827C25.8849 7.79981 23.4868 12.3702 23.4868 17.4737C23.4868 22.5773 25.8849 27.1476 29.6668 30.2192Z"
                  fill="#FF5E00"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
          <div
            class="flex items-center justify-between"
          >
            <p
              class="text-foreground-100 font-medium text-2xs opacity-50"
            >
              .... 
              9888
            </p>
            <p
              class="text-foreground-100 font-medium text-2xs opacity-50"
            />
          </div>
        </div>
      </div>
      <div
        aria-label="card-item-2"
        class="hover:cursor-pointer"
      >
        <div
          class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex h-[88px] w-full min-w-[111px] flex-col justify-between rounded-[6px] border-0 p-2 bg-background-500"
          tabindex="-1"
        >
          <div
            class="flex gap-3 p-0"
          >
            <div
              class="flex flex-1 items-center justify-between"
            >
              <p
                class="text-foreground-100 font-medium text-xs"
              >
                Savings
              </p>
              <svg
                class="w-[25px] h-[15px]"
                fill="none"
                height="35"
                viewBox="0 0 59 35"
                width="59"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M41.0762 34.2205C50.791 34.2205 58.6663 26.723 58.6663 17.4744C58.6663 8.22576 50.791 0.728271 41.0762 0.728271C36.7224 0.728271 32.738 2.23419 29.6664 4.72876C26.5948 2.23426 22.6104 0.72838 18.2566 0.72838C8.54186 0.72838 0.666504 8.22586 0.666504 17.4745C0.666504 26.7231 8.54186 34.2206 18.2566 34.2206C22.6105 34.2206 26.5949 32.7147 29.6665 30.2201C32.7381 32.7146 36.7224 34.2205 41.0762 34.2205Z"
                  fill="#ED0006"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M29.6665 30.2197C33.4484 27.1482 35.8465 22.5778 35.8465 17.4742C35.8465 12.3707 33.4484 7.80033 29.6665 4.72879C32.7381 2.23413 36.7226 0.728149 41.0766 0.728149C50.7913 0.728149 58.6667 8.22563 58.6667 17.4742C58.6667 26.7229 50.7913 34.2203 41.0766 34.2203C36.7226 34.2203 32.7381 32.7144 29.6665 30.2197Z"
                  fill="#F9A000"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M29.6668 30.2192C33.4488 27.1476 35.8469 22.5773 35.8469 17.4737C35.8469 12.3702 33.4488 7.79981 29.6668 4.72827C25.8849 7.79981 23.4868 12.3702 23.4868 17.4737C23.4868 22.5773 25.8849 27.1476 29.6668 30.2192Z"
                  fill="#FF5E00"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
          <div
            class="flex items-center justify-between"
          >
            <p
              class="text-foreground-100 font-medium text-2xs opacity-50"
            >
              .... 
              9357
            </p>
            <p
              class="text-foreground-100 font-medium text-2xs opacity-50"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MyCards component Should render snapshot with accounts is empty. 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-wrap gap-4"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-wrap gap-4"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MyCards component Should renders cards correctly with empty type and accountNumber 1`] = `
<DocumentFragment>
  <div
    class="flex flex-wrap gap-4"
  >
    <div
      aria-label="card-item-0"
      class="hover:cursor-pointer"
    >
      <div
        class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex h-[88px] w-full min-w-[111px] flex-col justify-between rounded-[6px] border-0 p-2 bg-secondary-300"
        tabindex="-1"
      >
        <div
          class="flex gap-3 p-0"
        >
          <div
            class="flex flex-1 items-center justify-between"
          >
            <p
              class="text-foreground-200 font-medium text-xs"
            />
            <svg
              class="w-[25px] h-[15px]"
              fill="none"
              height="35"
              viewBox="0 0 59 35"
              width="59"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M41.0762 34.2205C50.791 34.2205 58.6663 26.723 58.6663 17.4744C58.6663 8.22576 50.791 0.728271 41.0762 0.728271C36.7224 0.728271 32.738 2.23419 29.6664 4.72876C26.5948 2.23426 22.6104 0.72838 18.2566 0.72838C8.54186 0.72838 0.666504 8.22586 0.666504 17.4745C0.666504 26.7231 8.54186 34.2206 18.2566 34.2206C22.6105 34.2206 26.5949 32.7147 29.6665 30.2201C32.7381 32.7146 36.7224 34.2205 41.0762 34.2205Z"
                fill="#ED0006"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M29.6665 30.2197C33.4484 27.1482 35.8465 22.5778 35.8465 17.4742C35.8465 12.3707 33.4484 7.80033 29.6665 4.72879C32.7381 2.23413 36.7226 0.728149 41.0766 0.728149C50.7913 0.728149 58.6667 8.22563 58.6667 17.4742C58.6667 26.7229 50.7913 34.2203 41.0766 34.2203C36.7226 34.2203 32.7381 32.7144 29.6665 30.2197Z"
                fill="#F9A000"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M29.6668 30.2192C33.4488 27.1476 35.8469 22.5773 35.8469 17.4737C35.8469 12.3702 33.4488 7.79981 29.6668 4.72827C25.8849 7.79981 23.4868 12.3702 23.4868 17.4737C23.4868 22.5773 25.8849 27.1476 29.6668 30.2192Z"
                fill="#FF5E00"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
        <div
          class="flex items-center justify-between"
        >
          <p
            class="text-foreground-200 font-medium text-2xs opacity-100"
          >
            .... 
          </p>
          <p
            class="text-foreground-200 font-medium text-2xs opacity-100"
          />
        </div>
      </div>
    </div>
    <div
      aria-label="card-item-1"
      class="hover:cursor-pointer"
    >
      <div
        class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex h-[88px] w-full min-w-[111px] flex-col justify-between rounded-[6px] border-0 p-2 bg-background-500"
        tabindex="-1"
      >
        <div
          class="flex gap-3 p-0"
        >
          <div
            class="flex flex-1 items-center justify-between"
          >
            <p
              class="text-foreground-100 font-medium text-xs"
            />
            <svg
              class="w-[25px] h-[15px]"
              fill="none"
              height="35"
              viewBox="0 0 59 35"
              width="59"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M41.0762 34.2205C50.791 34.2205 58.6663 26.723 58.6663 17.4744C58.6663 8.22576 50.791 0.728271 41.0762 0.728271C36.7224 0.728271 32.738 2.23419 29.6664 4.72876C26.5948 2.23426 22.6104 0.72838 18.2566 0.72838C8.54186 0.72838 0.666504 8.22586 0.666504 17.4745C0.666504 26.7231 8.54186 34.2206 18.2566 34.2206C22.6105 34.2206 26.5949 32.7147 29.6665 30.2201C32.7381 32.7146 36.7224 34.2205 41.0762 34.2205Z"
                fill="#ED0006"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M29.6665 30.2197C33.4484 27.1482 35.8465 22.5778 35.8465 17.4742C35.8465 12.3707 33.4484 7.80033 29.6665 4.72879C32.7381 2.23413 36.7226 0.728149 41.0766 0.728149C50.7913 0.728149 58.6667 8.22563 58.6667 17.4742C58.6667 26.7229 50.7913 34.2203 41.0766 34.2203C36.7226 34.2203 32.7381 32.7144 29.6665 30.2197Z"
                fill="#F9A000"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M29.6668 30.2192C33.4488 27.1476 35.8469 22.5773 35.8469 17.4737C35.8469 12.3702 33.4488 7.79981 29.6668 4.72827C25.8849 7.79981 23.4868 12.3702 23.4868 17.4737C23.4868 22.5773 25.8849 27.1476 29.6668 30.2192Z"
                fill="#FF5E00"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
        <div
          class="flex items-center justify-between"
        >
          <p
            class="text-foreground-100 font-medium text-2xs opacity-50"
          >
            .... 
          </p>
          <p
            class="text-foreground-100 font-medium text-2xs opacity-50"
          />
        </div>
      </div>
    </div>
    <div
      aria-label="card-item-2"
      class="hover:cursor-pointer"
    >
      <div
        class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none flex h-[88px] w-full min-w-[111px] flex-col justify-between rounded-[6px] border-0 p-2 bg-background-500"
        tabindex="-1"
      >
        <div
          class="flex gap-3 p-0"
        >
          <div
            class="flex flex-1 items-center justify-between"
          >
            <p
              class="text-foreground-100 font-medium text-xs"
            />
            <svg
              class="w-[25px] h-[15px]"
              fill="none"
              height="35"
              viewBox="0 0 59 35"
              width="59"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M41.0762 34.2205C50.791 34.2205 58.6663 26.723 58.6663 17.4744C58.6663 8.22576 50.791 0.728271 41.0762 0.728271C36.7224 0.728271 32.738 2.23419 29.6664 4.72876C26.5948 2.23426 22.6104 0.72838 18.2566 0.72838C8.54186 0.72838 0.666504 8.22586 0.666504 17.4745C0.666504 26.7231 8.54186 34.2206 18.2566 34.2206C22.6105 34.2206 26.5949 32.7147 29.6665 30.2201C32.7381 32.7146 36.7224 34.2205 41.0762 34.2205Z"
                fill="#ED0006"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M29.6665 30.2197C33.4484 27.1482 35.8465 22.5778 35.8465 17.4742C35.8465 12.3707 33.4484 7.80033 29.6665 4.72879C32.7381 2.23413 36.7226 0.728149 41.0766 0.728149C50.7913 0.728149 58.6667 8.22563 58.6667 17.4742C58.6667 26.7229 50.7913 34.2203 41.0766 34.2203C36.7226 34.2203 32.7381 32.7144 29.6665 30.2197Z"
                fill="#F9A000"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M29.6668 30.2192C33.4488 27.1476 35.8469 22.5773 35.8469 17.4737C35.8469 12.3702 33.4488 7.79981 29.6668 4.72827C25.8849 7.79981 23.4868 12.3702 23.4868 17.4737C23.4868 22.5773 25.8849 27.1476 29.6668 30.2192Z"
                fill="#FF5E00"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
        <div
          class="flex items-center justify-between"
        >
          <p
            class="text-foreground-100 font-medium text-2xs opacity-50"
          >
            .... 
          </p>
          <p
            class="text-foreground-100 font-medium text-2xs opacity-50"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
